src/ows/src/main/java/org/geoserver/ows/kvp/ViewParamsKvpParser.java

```diff
@@ -36,23 +48,61 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 
     @Override
     public Object parse(String value) throws Exception {
-        List<Object> ret = new ArrayList<>();
-        List<KvpParser> parsers =
-                GeoServerExtensions.extensions(KvpParser.class, applicationContext);
-        KvpParser formatOptionsParser = null;
-        for (KvpParser parser : parsers) {
-            if (parser.getKey().equalsIgnoreCase("format_options")) {
-                formatOptionsParser = parser;
-                break;
-            }
+        ViewParamsFormatParser formatParser = getRequestFormatParser();
+        LOGGER.log(Level.FINE, "Using selected format parser: {0}", formatParser);
+        return formatParser.parse(value);
+    }
+
+    private ViewParamsFormatParser getRequestFormatParser() {
+        List<ViewParamsFormatParser> formatParsers =
+                GeoServerExtensions.extensions(ViewParamsFormatParser.class, applicationContext);
+        String formatParserParameterValue = getFormatParserParameterValue();
+        LOGGER.log(Level.FINE, "Using format parser identifier: {0}", formatParserParameterValue);
+        return formatParsers.stream()
+                .filter(fp -> fp.getIdentifier().equals(formatParserParameterValue))
+                .findFirst()
+                .get();
```

Comment:
This one will fail if ``viewParamsFormat`` contains a bogus value. The end user should see a ServiceException with the right code, ``InvalidParameterValue``, and locator, ``viewParamsFormat``, but I believe with the current  code will end up returning a generic error instead.

Please add a test covering invalid values too, check the right OGC exception is returned, and see the ``ServiceException`` class and its ``INVALID_PARAMETER_VALUE`` constant.

Reply:
Applied.